projects/js-packages/partner-coupon/components/redeem-partner-coupon-pre-connection/test/component.jsx

```diff
@@ -0,0 +1,250 @@
+	},
+};
+const requiredProps = {
+	apiNonce: 'NONCE',
+	apiRoot: 'ROOT',
+	connectionStatus: { isRegistered: true },
+	partnerCoupon: partnerCoupon,
+	assetBaseUrl: 'PATH',
+	registrationNonce: 'NONCE',
+	siteRawUrl: 'example.com',
+	tracksUserData: true,
+	analytics: analytics,
+};
+
+let locationAssignSpy;
+let recordEventStub;
+let storeSelect;
+let stubGetConnectionStatus;
+
+describe( 'RedeemPartnerCouponPreConnection - component', () => {
+	before( () => {
+		renderHook( () => useSelect( select => ( storeSelect = select( CONNECTION_STORE_ID ) ) ) );
+
+		locationAssignSpy = sinon.spy();
+		recordEventStub = sinon.stub( analytics.tracks, 'recordEvent' );
+		stubGetConnectionStatus = sinon.stub( storeSelect, 'getConnectionStatus' );
+	} );
+
+	beforeEach( () => {
+		// Spy on location.assign, so we don't get breaking errors when
+		// we trigger click events on buttons/links.
+		delete window.location;
+		window.location = { assign: locationAssignSpy };
+
+		stubGetConnectionStatus.returns( {} );
+	} );
+
+	afterEach( () => {
+		window.location = location;
+		locationAssignSpy.resetHistory();
+
+		recordEventStub.reset();
+		stubGetConnectionStatus.reset();
+	} );
+
+	after( () => {
+		recordEventStub.restore();
+		stubGetConnectionStatus.restore();
+	} );
+
+	it( 'partner is shown', () => {
+		render( <RedeemPartnerCouponPreConnection { ...requiredProps } /> );
+
+		expect( screen.getAllByText( 'Welcome to Jetpack Company name traveler!' ) ).to.exist;
+	} );
+
+	it( 'description is shown with product name', () => {
+		render( <RedeemPartnerCouponPreConnection { ...requiredProps } /> );
+
+		expect(
+			screen.getAllByText(
+				'Redeem your coupon and get started with Awesome Product for free the first year!'
+			)
+		).to.exist;
+	} );
+
+	it( 'features are shown', () => {
+		render( <RedeemPartnerCouponPreConnection { ...requiredProps } /> );
+
+		// eslint-disable-next-line no-unused-vars
+		for ( const [ key, feature ] of Object.entries( partnerCoupon.product.features ) ) {
+			expect( screen.getAllByText( feature ) ).to.exist;
+		}
+	} );
+
+	it( 'set up and redeem button - is shown for new sites', () => {
+		stubGetConnectionStatus.returns( { isRegistered: false, isUserConnected: false } );
+
+		const props = {
+			...requiredProps,
+			connectionStatus: {
+				hasConnectedOwner: false,
+			},
+		};
+
+		const { container } = render( <RedeemPartnerCouponPreConnection { ...props } /> );
+
+		expect(
+			screen.getByRole( 'button', {
+				name: 'Set up & redeem Awesome Product',
+			} )
+		).to.exist;
+
+		// We use querySelector because using typical screen.* selectors will give an
+		// error if the component doesn't exist; and we specifically want to ensure
+		// it doesn't exist while displaying the "Set up button".
+		expect( container.querySelector( 'button[aria-label="Redeem Awesome Product"]' ) ).to.not.exist;
+	} );
+
+	it( 'set up and redeem button - is shown for userless connections', () => {
```

Comment:
The test names should conform to the way describe() + it() are meant to be used.

For example, this test will be read as:
`RedeemPartnerCouponPreConnection - component set up and redeem button - is shown for userless connections`
IMO it should be switched to the more typical:
`describe( 'RedeemPartnerCouponPreConnection' ) + it( 'shows the "set up and redeem" button for userless connections' )`
so it reads like this:
`RedeemPartnerCouponPreConnection shows the "set up and redeem" button for userless connections`
(this is the way the test will be listed as when it fails, for example)

Reply:
That totally makes sense. Thank you for suggesting it! 🙇